contracts/rewards/StableUsnMaker.sol

```diff
@@ -0,0 +1,122 @@
+        stableSwap = ISwap(_stableSwap);
+        lpMaker = _lpMaker;
+        usn = _usn;
+        usdt = _usdt;
+        usdc = _usdc;
+    }
+
+    function setStableSwap(ISwap _stableSwap) public onlyOwner {
+        stableSwap = _stableSwap;
+
+        emit LogSetStableSwap(address(stableSwap), address(_stableSwap));
+    }
+
+    function setLPMaker(address _lpMaker) public onlyOwner {
+        address oldlpMaker;
+        oldlpMaker = lpMaker;
+        lpMaker = _lpMaker;
+
+        emit LogSetLPMaker(oldlpMaker, _lpMaker);
+    }
+
+    // C6: It's not a fool proof solution, but it prevents flash loans, so here it's ok to use tx.origin
+    modifier onlyEOA() {
+        // Try to make flash-loan exploit harder to do by only allowing externally owned addresses.
+        require(msg.sender == tx.origin, "StableUsnMaker: must use EOA");
+        _;
+    }
+
+    function withdrawStableTokenFees() public onlyEOA {
+        // Withdraw admin fees from the Stableswap Pool to stable tokens
+        stableSwap.withdrawAdminFees();
+
+        emit LogWithdrawFees();
+    }
+
+    // This onyl works for USDC and USDT. If we have more tokens, need to make this more dynamic
+    function swapStableTokensToUsn() public onlyEOA {
+        address[2] memory stableTokensToSwapToUsn = [usdc, usdt];
+        for (uint256 i = 0; i < stableTokensToSwapToUsn.length; i++) {
+            IERC20(stableTokensToSwapToUsn[i]).approve(
+                address(stableSwap),
+                IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this))
+            );
+
+            uint256 stableTokenAmount = IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this));
+
+            stableSwap.swap(
+                stableSwap.getTokenIndex(stableTokensToSwapToUsn[i]),
+                stableSwap.getTokenIndex(usn),
+                stableTokenAmount,
+                0,
+                block.timestamp + 60
+            );
+
+            emit LogSwapStableTokenToUsn(stableTokensToSwapToUsn[i], stableTokenAmount);
+        }
+    }
+
+    function sendUsnToLPMaker() public onlyEOA {
+        // Check the balanceOf converted USN and send to RDT for dishing up
+        uint256 usnAmount = IERC20(usn).balanceOf(address(this));
+        require(usnAmount > 0, "StableUsnMaker: no Usn to send");
+        IERC20(usn).safeTransfer(lpMaker, usnAmount);
+
+        emit LogUsnSentToLPMaker(usnAmount);
+    }
+
+    // Emergency Withdraw function
+    function reclaimTokens(address token, uint256 amount, address payable to) public onlyOwner {
+        if (token == address(0)) {
+            to.transfer(amount);
+        } else {
+            IERC20(token).safeTransfer(to, amount);
+        }
+    }
+
+    function convertStables() external onlyEOA {
+        // Withdraw admin fees from the stableswap pool to stable tokens
+        withdrawStableTokenFees();
+
+        // For each stable token, we need to convert it to usn via the stableswap
+        swapStableTokensToUsn();
```

Comment:
IMO better to remove this step completely and send all the Fee Tokens into LPMaker

Reply:
closing this based on discussion in discord